### ISSUE TITLE
java.lang.UnsatisfiedLinkError: C:\Windows\Temp\jna--703330221\jna8101769954861517871.dll

### ISSUE THREAD

**Issue by pinonpierre** [Waffle/waffle#696]
Hi,

One customer of our application have an issue but I cannot reproduce the error.

**Environment:**

- OpenJdk 11
- Waffle 1.8.1
- Jetty 9.4.14.v20181114
- Chrome
- Windows Server 2016

Seem to not happen before in Java Oracle 8

**Error:**

```
java.lang.UnsatisfiedLinkError: C:\Windows\Temp\jna--703330221\jna8101769954861517871.dll: Can't find dependent libraries 
at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method) 
at java.base/java.lang.ClassLoader$NativeLibrary.load(Unknown Source) 
at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(Unknown Source) 
at java.base/java.lang.ClassLoader.loadLibrary0(Unknown Source) 
at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source) 
at java.base/java.lang.Runtime.load0(Unknown Source) 
at java.base/java.lang.System.load(Unknown Source) 
at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:851) 
at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:826) 
at com.sun.jna.Native.<clinit>(Native.java:140) 
at com.sun.jna.Pointer.<clinit>(Pointer.java:41) 
at com.sun.jna.Structure.<clinit>(Structure.java:2078) 
at waffle.windows.auth.impl.WindowsCredentialsHandleImpl.initialize(WindowsCredentialsHandleImpl.java:82) 
at waffle.windows.auth.impl.WindowsAuthProviderImpl.acceptSecurityToken(WindowsAuthProviderImpl.java:117) 
at waffle.servlet.spi.NegotiateSecurityFilterProvider.doFilter(NegotiateSecurityFilterProvider.java:144) 
at waffle.servlet.spi.SecurityFilterProviderCollection.doFilter(SecurityFilterProviderCollection.java:148) 
at waffle.servlet.NegotiateSecurityFilter.doFilter(NegotiateSecurityFilter.java:123) 
at com.sma.sm.app.SmSecurityFilter.doFilter(SmSecurityFilter.java:94) 
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637) 
at org.eclipse.jetty.servlets.QoSFilter.doFilter(QoSFilter.java:201) 
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637) 
at com.sma.sm.app.HeaderSetterFilter.doFilter(HeaderSetterFilter.java:44) 
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637) 
at org.eclipse.jetty.servlets.DoSFilter.doFilterChain(DoSFilter.java:480) 
at org.eclipse.jetty.servlets.DoSFilter.doFilter(DoSFilter.java:325) 
at org.eclipse.jetty.servlets.DoSFilter.doFilter(DoSFilter.java:295) 
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637) 
at com.sma.sm.app.UrlFilter.doFilter(UrlFilter.java:68) 
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637) 
at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533) 
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) 
at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) 
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) 
at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190) 
at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595) 
at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188) 
at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253) 
at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168) 
at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473) 
at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564) 
at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166) 
at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155) 
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) 
at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:61) 
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) 
at org.eclipse.jetty.server.Server.handle(Server.java:530) 
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347) 
at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256) 
at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279) 
at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102) 
at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124) 
at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247) 
at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140) 
at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708) 
at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626) 
at java.base/java.lang.Thread.run(Unknown Source) 
```

Do you have some idea to solve this issue please?

**Comment by dblock** [comment]
There's likely a Windows CRT dependency missing or something like that. If you have a computer where it happens I would run `depends C:\Windows\Temp\jna--703330221\jna8101769954861517871.dll`, using http://www.dependencywalker.com and it should produce the same error.

**Comment by pinonpierre** [comment]
Hi,

Seems to happen only on Windows 2016.
Bellow the reports on my functionnal machine:

[jna4789867345758265034_windows_10.zip](https://github.com/Waffle/waffle/files/3131428/jna4789867345758265034_windows_10.zip)

And on the Windows Server 2016 where the issue occurs:

[jna4789867345758265034_windows_server_2016.zip](https://github.com/Waffle/waffle/files/3131430/jna4789867345758265034_windows_server_2016.zip)

It seems in relations with these libraries:
- API-MS-WIN-CORE-PROFILE-L1-1-0.DLL
- API-MS-WIN-CORE-PROCESSTHREADS-L1-1-0.DLL
- API-MS-WIN-CORE-SYSINFO-L1-1-0.DLL
- API-MS-WIN-CORE-ERRORHANDLING-L1-1-0.DLL
- MSVCR100.DLL
- API-MS-WIN-CORE-RTLSUPPORT-L1-1-0.DLL

I tried to install the last "Redistribuable Visual C++ for Visual Studio" but without any success

When I search these libraries, I can find them in the "C:\Windows" directory!

**Comment by pinonpierre** [comment]
I didn't see that the Maven GroupId has been changed.

The issues seems to be fixed since the 1.8.3 version and work correctly on the last 1.9.1 version

Thanks!

**Comment by dblock** [comment]
In your other comment you mentioned that there was a dependency on MSVCR100.DLL (Visual Studio CRT). That shouldn't happen and I suspect it has changed in 1.8.3 to correctly depend on the windows CRT, which is MSVCRT.DLL (version-less and exists on all machines). Welcome (back) to DLL hell.
